FIAT/reference_element.py

```diff
@@ -1751,15 +1697,16 @@ def is_hypercube(cell):
 
 def flatten_reference_cube(ref_el):
     """This function flattens a Tensor Product hypercube to the corresponding UFC hypercube"""
-    flattened_cube = {2: UFCQuadrilateral(), 3: UFCHexahedron()}
     if numpy.sum(ref_el.get_dimension()) <= 1:
         # Just return point/interval cell arguments
         return ref_el
     else:
         # Handle cases where cell is a quad/cube constructed from a tensor product or
         # an already flattened element
-        if is_hypercube(ref_el):
-            return flattened_cube[numpy.sum(ref_el.get_dimension())]
+        if isinstance(ref_el, TensorProductCell):
+            return Hypercube(numpy.sum(ref_el.get_dimension()), ref_el)
```

Comment:
I think this gives you the integer dimension
```suggestion
            return Hypercube(ref_el.get_spatial_dimension(), ref_el)
```